internal/storage/bucket_handle_test.go

```diff
@@ -145,43 +146,83 @@ func (t *BucketHandleTest) TestNewReaderMethodWithInvalidGeneration() {
 }
 
 func (t *BucketHandleTest) TestDeleteObjectMethodWithValidObject() {
-	error := t.bucketHandle.DeleteObject(context.Background(),
+	err := t.bucketHandle.DeleteObject(context.Background(),
 		&gcs.DeleteObjectRequest{
 			Name:                       TestObjectName,
 			Generation:                 TestObjectGeneration,
 			MetaGenerationPrecondition: nil,
 		})
 
-	AssertEq(nil, error)
+	AssertEq(nil, err)
 }
 
 func (t *BucketHandleTest) TestDeleteObjectMethodWithMissingObject() {
-	error := t.bucketHandle.DeleteObject(context.Background(),
+	err := t.bucketHandle.DeleteObject(context.Background(),
 		&gcs.DeleteObjectRequest{
 			Name:                       missingObjectName,
 			Generation:                 TestObjectGeneration,
 			MetaGenerationPrecondition: nil,
 		})
 
-	AssertEq("storage: object doesn't exist", error.Error())
+	AssertEq("storage: object doesn't exist", err.Error())
 }
 
 func (t *BucketHandleTest) TestStatObjectMethodWithValidObject() {
-	_, error := t.bucketHandle.StatObject(context.Background(),
+	_, err := t.bucketHandle.StatObject(context.Background(),
 		&gcs.StatObjectRequest{
 			Name: TestObjectName,
 		})
 
-	AssertEq(nil, error)
+	AssertEq(nil, err)
 }
 
 func (t *BucketHandleTest) TestStatObjectMethodWithMissingObject() {
 	var notfound *gcs.NotFoundError
 
-	_, error := t.bucketHandle.StatObject(context.Background(),
+	_, err := t.bucketHandle.StatObject(context.Background(),
 		&gcs.StatObjectRequest{
 			Name: missingObjectName,
 		})
 
-	AssertTrue(errors.As(error, &notfound))
+	AssertTrue(errors.As(err, &notfound))
+}
+
+func (t *BucketHandleTest) TestCopyObjectMethodWithValidObject() {
+	_, err := t.bucketHandle.CopyObject(context.Background(),
+		&gcs.CopyObjectRequest{
+			SrcName:                       TestObjectName,
+			DstName:                       dstObjectName,
+			SrcGeneration:                 TestObjectGeneration,
+			SrcMetaGenerationPrecondition: nil,
+		})
+
+	AssertEq(nil, err)
+}
+
+func (t *BucketHandleTest) TestCopyObjectMethodWithMissingObject() {
+	var notfound *gcs.NotFoundError
+
+	_, err := t.bucketHandle.CopyObject(context.Background(),
+		&gcs.CopyObjectRequest{
+			SrcName:                       missingObjectName,
+			DstName:                       dstObjectName,
+			SrcGeneration:                 TestObjectGeneration,
+			SrcMetaGenerationPrecondition: nil,
+		})
+
+	AssertTrue(errors.As(err, &notfound))
+}
+
+func (t *BucketHandleTest) TestCopyObjectMethodWithInvalidGeneration() {
+	var notfound *gcs.NotFoundError
+
+	_, err := t.bucketHandle.CopyObject(context.Background(),
+		&gcs.CopyObjectRequest{
+			SrcName:                       TestObjectName,
+			DstName:                       dstObjectName,
+			SrcGeneration:                 222, // other than TestObjectGeneration, doesn't exist.
```

Comment:
make the comment more clear. Other than testObjectGeneration, no other generation exists.